tests/test_coadd_outputs.py

```diff
@@ -0,0 +1,364 @@
+                coadd_coord = geom.SpherePoint(cat["coord_ra"].values[index]*geom.degrees,
+                                               cat["coord_dec"].values[index]*geom.degrees)
+                summed_throughput = np.zeros(wavelengths.shape, dtype=np.float64)
+                weight_sum = 0.0
+                for rec in inputs.subsetContaining(coadd_coord, includeValidPolygon=True):
+                    det_pos = rec.getWcs().skyToPixel(coadd_coord)
+                    det_trans = rec.getTransmissionCurve()
+                    weight = rec.get("weight")
+                    summed_throughput += det_trans.sampleAt(det_pos, wavelengths)*weight
+                    weight_sum += weight
+                if weight_sum == 0.0:
+                    continue
+                summed_throughput /= weight_sum
+                coadd_pos = wcs.skyToPixel(coadd_coord)
+                coadd_throughput = transmission_curve.sampleAt(coadd_pos, wavelengths)
+                np.testing.assert_array_almost_equal(coadd_throughput, summed_throughput)
+                n_tested += 1
+            self.assertGreater(n_tested, 5)
+
+    def test_mask_planes_exist(self):
+        """Test that the input mask planes have been added."""
+        for data_id in DATA_IDS:
+            mask = self.butler.get("calexp.mask", data_id)
+            self.assertIn("CROSSTALK", mask.getMaskPlaneDict())
+            self.assertIn("NOT_DEBLENDED", mask.getMaskPlaneDict())
+
+    # Expected to fail until DM-5174 is fixed.
+    @unittest.expectedFailure
+    def test_masks_removed(self):
+        """Test that certain mask planes have been removed from the coadds.
+
+        This is expected to fail until DM-5174 is fixed.
+        """
+        for band in self._bands:
+            mask = self.butler.get("deepCoadd_calexp.mask", band=band, tract=self._tract, patch=self._patch)
+            self.assertNotIn("CROSSTALK", mask.getMaskPlaneDict())
+            self.assertNotIn("NOT_DEBLENDED", mask.getMaskPlaneDict())
+
+    def test_warp_inputs(self):
+        """Test that the warps have the correct inputs."""
+        skymap = self.butler.get("skyMap")
+        tract_info = skymap[self._tract]
+        for warp_type in ["directWarp", "psfMatchedWarp"]:
+            datasets = set(self.butler.registry.queryDatasets(f"deepCoadd_{warp_type}"))
+            # We only need to test one dataset
+            dataset = list(datasets)[0]
+
+            warp = self.butler.getDirect(dataset)
+            self.assertEqual(warp.wcs, tract_info.wcs)
+            inputs = warp.getInfo().getCoaddInputs()
+            self.assertEqual(len(inputs.visits), 1)
+            visit_record = inputs.visits[0]
+            self.assertEqual(visit_record.getWcs(), warp.wcs)
+            self.assertEqual(visit_record.getBBox(), warp.getBBox())
+            self.assertGreater(len(inputs.ccds), 0)
+
+            wcs_cat = self.butler.get(
+                "jointcalSkyWcsCatalog",
```

Comment:
In principle, this could change, but I guess we can't introspect what the external calib names are set to be for this repo, so they will just have to be updated in tandem?

Reply:
I considered having it check the config (which it could do!) and test that it's okay.  And after some consideration I thought that this is something that we have explicitly set in the `ci_hsc_gen3` pipeline, and therefore want to explicitly check.  If the pipeline changes, then the checks of the pipeline should be changed as well.  But if you think it would be better to have config-reading logic here that could be done, but I _think_ it would be less robust.

Reply:
Nah, I think your reasoning is sound.  It makes sense for this to explicitly check the hard-coded pipeline (and churn in this regard is likely be too infrequent for us to worry too much about the "best" option here!)